clang_github_tracker/services.py

```
@@ -0,0 +1,270 @@
+    return inserted, updated
+
+
+def upsert_issue_items_batch(
+    rows: Sequence[tuple[int, bool, datetime | None, datetime | None]],
+    *,
+    batch_size: int = DEFAULT_UPSERT_BATCH_SIZE,
+) -> tuple[int, int]:
+    """Batch upsert issue/PR rows by ``number``.
+
+    Duplicate ``number`` values merge: ``github_updated_at`` uses the latest
+    timestamp; ``github_created_at`` uses a later row’s value when non-None,
```

Ruff (0.15.9)

[warning] Docstring contains ambiguous `’` (RIGHT SINGLE QUOTATION MARK). Did you mean ``` (GRAVE ACCENT)?

(RUF002)